stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/suggestions/SuggestionListView.kt

```diff
@@ -0,0 +1,78 @@
+package io.getstream.chat.android.ui.suggestions
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.FrameLayout
+import androidx.core.view.isVisible
+import androidx.recyclerview.widget.ConcatAdapter
+import io.getstream.chat.android.client.models.Command
+import io.getstream.chat.android.client.models.User
+import io.getstream.chat.android.ui.databinding.StreamSuggestionListViewBinding
+
+internal class SuggestionListView : FrameLayout {
+
+    private val binding: StreamSuggestionListViewBinding = LayoutInflater.from(context).let {
+        StreamSuggestionListViewBinding.inflate(it, this)
+    }
+    private val mentionsAdapter: MentionsAdapter = MentionsAdapter { listener?.onMentionClick(it) }
+    private val commandsAdapter: CommandsAdapter = CommandsAdapter { listener?.onCommandClick(it) }
+
+    private var listener: OnSuggestionClickListener? = null
+
+    constructor(context: Context) : super(context)
+
+    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs)
+
+    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(
+        context,
+        attrs,
+        defStyleAttr
+    )
+
+    init {
+        binding.suggestionsRecyclerView.apply {
+            itemAnimator = null
+            adapter = ConcatAdapter(mentionsAdapter, commandsAdapter)
+        }
+    }
+
+    fun setSuggestions(suggestions: Suggestions) {
+        if (!binding.suggestionsCardView.isVisible) {
+            binding.suggestionsCardView.isVisible = true
+        }
+        when (suggestions) {
+            is Suggestions.MentionSuggestions -> {
+                mentionsAdapter.submitList(suggestions.users)
+                binding.commandsTitleTextView.isVisible = false
+            }
+            is Suggestions.CommandSuggestions -> {
+                commandsAdapter.submitList(suggestions.commands)
+                binding.commandsTitleTextView.isVisible = true
+            }
+        }
+    }
+
+    fun setOnSuggestionClickListener(listener: OnSuggestionClickListener) {
+        this.listener = listener
+    }
+
+    fun clearSuggestions() {
+        if (binding.suggestionsCardView.isVisible) {
+            commandsAdapter.submitList(emptyList())
+            mentionsAdapter.submitList(emptyList())
+            binding.suggestionsCardView.isVisible = false
+        }
+    }
+}
+
+public interface OnSuggestionClickListener {
```

Comment:
I'd do more nesting on this PR. I'd move classes that belong directly to a single other class (e.g. this listener, the sealed class below, the ViewHolder class in the other file). More clearly indicates their intended scope. And if they're used in multiple places, they can live in their own file.

Reply:
Might be a bit verbose at times but makes perfect sense

Reply:
Done

Reply:
I usually add more specific imports for these, so that their usages are still as short as if they were top level classes.